scripts/release.sh

```diff
@@ -71,13 +72,26 @@ if [ "$PRE_VERSION_HASH" = "$POST_VERSION_HASH" ]; then
   exit 1
 fi
 
+# read tags and push to active branch 
 echo "${GREEN}Pushing tags and version update commit to Github...${NC}"
 read_previous_commit_tags
 
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
 git push --set-upstream origin $BRANCH
 git push origin $TAGS
 
+# create temporary release branch from latest commit which is tagged
+# add an empty commit so it can be pushed to origin and PR'd then 
+# move back to previous branch so tags can be deleted with --undo
+# if necessary.
+echo "${GREEN}Creating temporary release branch...${NC}"
+BRANCHREF=$(git rev-parse --short HEAD)
+TEMP_BRANCH="release-${BRANCHREF}"
+git checkout -b $TEMP_BRANCH
+git commit --allow-empty -m "DS Release Bump"
```

Comment:
Do we need this commit? What will happen when we try merging this branch into master? Wouldn't it end up merging the whole release branch so far? Maybe we actually need to create a new branch off of master and cherry-pick the bump commit onto this one 😵

Reply:
Hmm, it would merge this branch into master but this branch should only have commits that were already in master. So it would only merge the version bumps when brought up to date with master no?

Reply:
Hmm, it would merge this branch into master but this branch should only have commits that were already in master. So it would only merge the version bumps when brought up to date with master no?

Reply:
Hmm, it would merge this branch into master but this branch should only have commits that were already in master. So it would only merge the version bumps when brought up to date with master no?

Reply:
The commit is so we can push this branch to origin, since it's empty.

Reply:
This temp branch is built off of the release branch, so it would have all previous commits in the release that would become merge conflicts that need to be resolved. That was a shortcoming of the previous way of doing it—if we had cherry-picked releases, we'd often get merge conflicts. I guess either way the issue can be resolved in the pull request that is made.

Reply:
Ah, yes that's true, but by cherry picking to the release branch can avoid that, good point.. I'll see what I can do to update for that.

Reply:
Doing it that way means we would need to merge the version bump before we could release which kinda stinks.